slick/src/main/scala/slick/jdbc/JdbcDataSource.scala

```diff
@@ -173,6 +175,26 @@ object HikariCPJdbcDataSource extends JdbcDataSourceFactory {
     val ds = new HikariDataSource(hconf)
     new HikariCPJdbcDataSource(ds, hconf)
   }
+
+  def parseDatabaseUrl(c: Config, callback:(Option[String], Option[String]) => Unit): String = {
+    val urlFromConfig = c.getStringOr("url", null)
+    if (urlFromConfig ne null) {
+      if (urlFromConfig.startsWith("jdbc")) {
+        callback(c.getStringOpt("user"), c.getStringOpt("password"))
+        urlFromConfig
+      } else {
+        val dbUri = new URI(urlFromConfig)
```

Comment:
The problem I'm seeing is that you go into this block with any string that doesn't start with jdbc. This here could fail with a `URISyntaxException`, which may actually be OK, but `dbUri.getUserInfo()` and similar could easily fail with a NPE.
You could either wrap this block in a try catch or, a bit nicer IMO, use a regular expression (as seen in Play).